lib/sql/rows.go

```diff
@@ -0,0 +1,34 @@
+package sql
+
+import "database/sql"
+
+func RowsToObjects(rows *sql.Rows) ([]map[string]any, error) {
+	defer rows.Close()
+
+	columns, err := rows.Columns()
+	if err != nil {
+		return nil, err
+	}
+
+	var objects []map[string]any
+	for rows.Next() {
```

Comment:
You'll need to check if `rows.Err()` is not nil after iterating through `rows.Next()`.

Reply:
https://github.com/golang/go/blob/356ba0f06586a833cd8de9c04af0d2adddf95851/src/database/sql/sql.go#L3122

Reply:
sg